### PR TITLE
Fsaverage5 fetcher, example separation, documentation 

### PR DESCRIPTION
**Pull request by Joaoloula**
- Separation of the example into an atlas one and a statistical map one;
- Creation of a fetcher for fsaverage5 (and removal of this data from the NKI fetcher);
- Documentation for the datasets and functions;
- I also created jupyter notebooks for the examples, but I'm not sure how to add them to the nilearn doc.

Yeah :)

### CODE REVIEW

File: examples/01_plotting/plot_surf_atlas.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"""
+Demonstrate basic loading and plotting of a cortical surface atlas.
+===================================================================
+The Destrieux parcellation (Destrieux et al, 2010) in fsaverage5 space as
+distributed with Freesurfer is used as the chosen atlas.
+
+The :func:`nilearn.plotting.plot_surf_stat_map` function is used
```
Comment (juhuntenburg) [inline examples/01_plotting/plot_surf_atlas.py]: plot_surf_roi

File: examples/01_plotting/plot_surf_atlas.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+"""
+Demonstrate basic loading and plotting of a cortical surface atlas.
+===================================================================
+The Destrieux parcellation (Destrieux et al, 2010) in fsaverage5 space as
+distributed with Freesurfer is used as the chosen atlas.
```
Comment (juhuntenburg) [inline examples/01_plotting/plot_surf_atlas.py]: Maybe add the references mentioned below (Dale et al, 1999, Fischl et al 1999) here after Freesurfer

File: examples/01_plotting/plot_surf_atlas.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# Display Destrieux parcellation on fsaverage5 surface
+plotting.plot_surf_roi(fsaverage5_pial, roi_map=parcellation[0],
+                       hemi='left', view='lateral', bg_map=sulcal_depth_map,
+                       bg_on_data=True, darkness=.5, cmap='gist_ncar')
+
+# Display Destrieux parcellation on inflated fsaverage5 surface
+plotting.plot_surf_roi(fsaverage5_inflated, roi_map=parcellation[0],
+                       hemi='left', view='lateral', bg_map=sulcal_depth_map,
+                       bg_on_data=True, darkness=.5, cmap='gist_ncar')
+
```
Comment (juhuntenburg) [inline examples/01_plotting/plot_surf_atlas.py:57]: Maybe we can brainstorm what would be the best way to use a lut for the colouring in plot_surf_roi?

File: examples/01_plotting/plot_surf_stat_map.py [hunk 4]
```diff
@@ -106,31 +103,9 @@
...
-                            hemi='left', bg_map=sulcal_depth_map,
-                            bg_on_data=True, view='medial', threshold=.6)
-
-# changing the colormap and alpha
+# Displaying a thresholded stat map with a different colormap and transparency
 plotting.plot_surf_stat_map(fsaverage5_pial, stat_map=stat_map, hemi='left',
                             bg_map=sulcal_depth_map, bg_on_data=True,
                             cmap='Spectral', threshold=.6, alpha=.5)
 
-# saving plots to file
```
Comment (juhuntenburg) [inline examples/01_plotting/plot_surf_stat_map.py:130]: Why did you remove this?

File: nilearn/datasets/description/destrieux_surface.rst [hunk 5]
```diff
@@ -0,0 +1,20 @@
+Destrieux Surface
+
+
+Notes
+-----
+Anatomical parcellation of the cortex based on sulcal depth. This atlas is
+based on the fsaverage5 surface data.
```
Comment (juhuntenburg) [inline nilearn/datasets/description/destrieux_surface.rst]: Maybe: "Anatomical parcellation of the cortex based on sulco-gyral patterns (Destrieux et al, 2010) distributed with Freesurfer (Dale et al, 1999, Fischl et al, 1999) in fsaverage5 standard surface space."

File: nilearn/datasets/description/destrieux_surface.rst [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    :'annot_left': parcellation on left hemisphere in freesurfer annot format
+    :'annot_right': parcellation on left hemisphere in freesurfer annot format
+
+
+References
+----------
+
+Destrieux, C., et al. "A sulcal depth-based anatomical parcellation
+of the cerebral cortex." NeuroImage 47 (2009): S151.
+http://dx.doi.org/10.1016/s1053-8119(09)71561-7
```
Comment (juhuntenburg) [inline nilearn/datasets/description/destrieux_surface.rst:21]: Add Dale and Fischl references

File: nilearn/datasets/description/fsaverage5.rst [hunk 7]
```diff
@@ -0,0 +1,22 @@
+fsaverage5
+
+
+Notes
+-----
+Freesurfer fsaverage5 averaged subject surface data.
```
Comment (juhuntenburg) [inline nilearn/datasets/description/fsaverage5.rst]: Fsaverage5 standard surface as distributed with Freesurfer (Dale et al, 1999, Fischl et al, 1999)

File: nilearn/datasets/description/fsaverage5.rst [hunk 8]
```diff
@@ -0,0 +1,22 @@
+fsaverage5
+
+
+Notes
+-----
+Freesurfer fsaverage5 averaged subject surface data. 
+
+Content
+-------
+    :'pial_left': pial surface data for the left hemisphere
```
Comment (juhuntenburg) [inline nilearn/datasets/description/fsaverage5.rst]: I would say "pial surface mesh", and "inflated pial surface mesh"

File: nilearn/datasets/description/nki_enhanced_surface.rst [hunk 9]
```diff
@@ -1 +1,22 @@
+NKI enhanced surface
 
+
+Notes
+-----
+NKI enhanced resting-state dataset, preprocessed and projected to the
```
Comment (juhuntenburg) [inline nilearn/datasets/description/nki_enhanced_surface.rst:6]: I'll need to write a bit more about which data of eNKI this is, we should also include the link to the preprocessing pipeline

File: nilearn/datasets/func.py [hunk 10]
```diff
@@ -1805,11 +1769,85 @@ def fetch_surf_nki_enhanced(n_subjects=30, data_dir=None,
...
+
+    url: string, optional
+        Override download URL. Used for test only (or if you setup a mirror of
+        the data). Default: None
+
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        Dictionary-like object, the interest attributes are :
+         -
```
Comment (juhuntenburg) [inline nilearn/datasets/func.py]: include info

**Review by juhuntenburg — APPROVED** [review]
Looks fantastic! I made some small comments, mostly wording in the documentation. But I can also just merge and fix these myself. 
Yeah!

File: nilearn/datasets/description/nki_enhanced_surface.rst (hunk 9 above)
Comment (Joaoloula) [reply nilearn/datasets/description/nki_enhanced_surface.rst:6]: Great, could you do that?

File: examples/01_plotting/plot_surf_atlas.py (hunk 3 above)
Comment (Joaoloula) [reply examples/01_plotting/plot_surf_atlas.py:57]: If you're ok with it, we could perhaps push this to the PR as is for the moment and note in the comments that we'd like to address this as well as the whitespace and annot format questions, and ask the maintainers what they think would be the best ways of handling these things.

**Comment by Joaoloula** [comment]
So, except for the 2 comments I replied to, everything else should have been fixed by this last commit: let me know if you think I should change anything else!
